src/runtime_src/driver/include/experimental/xrt-next.h

```diff
@@ -0,0 +1,105 @@
+ * @sysfsPath:           Return string with full path for sysfs node
+ * @size:                Length of the return string
+ * Return:               0 or standard error number
+ *
+ * (For debug and profile usage only for now)
+ * The sysfs information is not accessible above XRT layer now. However, debug/profile
+ * needs information from sysfs (for example debug_ip_layout) to properly initialize
+ * xdp code, so this helper API can be used
+ */
+XCL_DRIVER_DLLESPEC int xclGetSysfsPath(xclDeviceHandle handle, const char* subdev,
+                                        const char* entry, char* sysfsPath, size_t size);
+
+/**
+ * Experimental APIs for reading debug and profile
+ *
+ * Warning: These APIs are experimental and can be
+ * changed or removed at any time. They should only
+ * be used by debug and profile code.
+ *
+ * @param handle the device handle
+ * @param info the xclDebugProfileDeviceInfo
+ * structure that this API will fill in as
+ * result
+ */
+XCL_DRIVER_DLLESPEC int xclGetDebugProfileDeviceInfo(xclDeviceHandle handle, xclDebugProfileDeviceInfo* info);
+
+/**
+  * xclMPD - Management Proxy Daemon API
+  *
+  * @handle:           Device handle
+  * @args:             software mailbox struct
+  *
+  * This API passes messages through the software channel of the userpf mailbox. The software mailbox struct
+  * has the following members:
+  * uint64_t flags:    reserved
+  * uint32_t *data:    message payload
+  * bool is_tx:        direction bit
+  * size_t sz:         when called, this indicates the size of the userspace buffer, upon return, it will
+  *                    be filled with the message payload size
+  * uint64_t id:       message id
+  *
+  * Returns 0 on success and nonzero on failure. errno will be set to EMSGSIZE when the passed userspace
+  * buffer is too small for the outbound message. This should only happen in the is_tx=true condition.
+  */
+XCL_DRIVER_DLLESPEC int xclMPD(xclDeviceHandle handle, struct drm_xocl_sw_mailbox *args);
+
+/**
+  * xclMPD - Management Service Daemon API
+  *
+  * @handle:           Device handle
+  * @args:             software mailbox struct
+  *
+  * This API passes messages through the software channel of the mgmtpf mailbox. The software mailbox struct
+  * has the following members:
+  * uint64_t flags:    reserved
+  * uint32_t *data:    message payload
+  * bool is_tx:        direction bit
+  * size_t sz:         when called, this indicates the size of the userspace buffer, upon return, it will
+  *                    be filled with the message payload size
+  * uint64_t id:       message id
+  *
+  * Returns 0 on success and nonzero on failure. errno will be set to EMSGSIZE when the passed userspace
+  * buffer is too small for the outbound message. This should only happen in the is_tx=true condition.
+  */
+XCL_DRIVER_DLLESPEC int xclMSD(xclDeviceHandle handle, struct drm_xocl_sw_mailbox *args);
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif
```

Comment:
want to also move xclP2pEnable(), xclMapMgmt() and xclOpenMgmt() from xrt.h into this file?

Reply:
Done